index.d.ts

```diff
@@ -1,18 +1,45 @@
 /**
- * Opens file(s) from disk.
+ * Properties shared by all `options` provided to file save and open operations
  */
-export function fileOpen<M extends boolean | undefined = false>(options?: {
-  /** Acceptable MIME types. [] */
-  mimeTypes?: string[];
-  /** Acceptable file extensions. Defaults to "". */
+export interface CoreFileOptions {
+  /** Acceptable file extensions. Defaults to [""]. */
   extensions?: string[];
   /** Suggested file description. Defaults to "". */
   description?: string;
-  /** Allow multiple files to be selected. Defaults to false. */
-  multiple?: M;
+  /** Acceptable MIME types. [] */
+  mimeTypes?: string[];
+}
+
+/**
+ * Properties shared by the _first_ `options` object provided to file save and
+ * open operations (any additional options objects provided to those operations
+ * cannot have these properties)
+ */
+export interface FirstCoreFileOptions extends CoreFileOptions {
   startIn?: WellKnownDirectory | FileSystemHandle;
   /** By specifying an ID, the user agent can remember different directories for different IDs. */
   id?: string;
+}
+
+/**
+ * The first `options` object passed to file save operations can also specify
+ * a filename
```

Comment:
This comes from spec?

Reply:
The fileName on the first `options` object gets used by the library as the "suggestedName" when it calls `window.showSaveFilePicker` (or as the "download" in the case of the legacy API); you can see it [here](https://github.com/GoogleChromeLabs/browser-fs-access/blob/main/src/fs-access/file-save.mjs#L63).

Reply:
I see. Shouldn't we use the filename of current `option` object instead @tomayac? That said, how is it even supposed to work for multiple files — doesn't that open the file save dialogs consecutively?

Reply:
I initially had roughly the same question (but left it in my mind, so I'm glad you asked it). I think we actually aren't in a case where we're saving multiple files, right? The scenario, I think, is that we're saving _one_ file, but passing multiple `options` objects along so that the file picker can allow choices between specific types when the file is saved (like in the case where you might be saving text, but can choose plain text, rich text, etc.). In other words, the additional `options` objects (beyond the first) are just specifying the things that get passed to `window.showSaveFilePicker` on the `types` array described [here](https://developer.mozilla.org/en-US/docs/Web/API/window/showSaveFilePicker). Does that answer the question?

Reply:
Oh nvm, I thought the `showSaveFilePicker()` call was still nested in the forEach loop (which wouldn't make sense, considering the `await` etc.). My bad!

Reply:
Yeah, it's a bit confusing at first sight, but makes sense in some scenarios. An IDE initially doesn't know if your text file will be a CSS (`text/css`), JS (`text/javascript`), or HTML (`text/html`) file for example.

Reply:
The `fileSave()` method initially didn't have exception handling, but Safari actually sort of has a cancel action, albeit is more for downloads per se as per a specific download. It's added now, analog to how it works with `options.setupLegacyCleanupAndRejection` in `fileOpen()` and `directoryOpen()`. 

<img width="490" alt="Screen Shot 2021-08-09 at 15 43 21" src="https://user-images.githubusercontent.com/145676/128716149-e5b9a561-07be-4e5a-b183-8c62ed4fea84.png">